target/linux/ath79/dts/qca9563_dlink_covr-p2500-a1.dts

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca956x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/mtd/partitions/uimage.h>
+
+/ {
+	compatible = "dlink,covr-p2500-a1", "qca,qca9563";
+	model = "D-Link COVR-P2500 A1";
+
+	aliases {
+		led-boot = &led_power_green;
+		led-failsafe = &led_power_red;
+		led-running = &led_power_green;
+		led-upgrade = &led_power_red;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		};
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&jtag_disable_pins>;
+
+		lan {
+			label = "green:lan";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_green: power_green {
+			label = "green:power";
+			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		wlan5g {
+			label = "green:wlan5g";
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0radio";
+		};
+
+		led_power_red: power_red {
+			label = "red:power";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "green:wlan2g";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1radio";
+		};
+	};
+
+	virtual_flash {
+		compatible = "mtd-concat";
+
+		devices = <&fwconcat0 &fwconcat1>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				compatible = "openwrt,uimage", "denx,uimage";
+				openwrt,ih-magic = <0x68737173>;
+				label = "firmware";
+				reg = <0x0 0x0>;
+			};
+		};
+	};
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x40000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			fwconcat0: partition@50000 {
+				label = "fwconcat0";
+				reg = <0x50000 0xe30000>;
+			};
+
+			partition@e80000 {
+				label = "loader";
+				reg = <0xe80000 0x10000>;
+				read-only;
+			};
+
+			fwconcat1: partition@e90000 {
+				label = "fwconcat1";
+				reg = <0xe90000 0x160000>;
+			};
+
+			art: partition@ff0000 {
+				label = "art";
+				reg = <0xff0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy-mask = <0>;
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+		qca,mib-poll-interval = <500>;
+		reset-gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
```

Comment:
The local ath driver doesn't support reset gpio. Please use `gpio-hog` to pull it high forever.

Reply:
wait, is this a regression introduced with DSA migration here? (does ath79 use DSA fpr the gigabit switches at all?)
I had introduced the `reset-gpios` declaration for DIR-842 and DIR-859, and it definitely worked for those devices a couple years ago (i.e. changing it to `GPIO_ACTIVE_HIGH` was confirmed to have the switch disabled)

Reply:
> wait, is this a regression introduced with DSA migration here? (does ath79 use DSA / `qca8k` for the gigabit switches at all?)

Currently, it is still a legacy swconfig driver. In
https://github.com/openwrt/openwrt/tree/main/target/linux/generic/files/drivers/net/phy


> I had introduced the `reset-gpios` declaration for DIR-842 and DIR-859, and it definitely worked for those devices a couple years ago (i.e. changing it to `GPIO_ACTIVE_HIGH` was confirmed to have the switch disabled)

No, that's because reset GPIO did not take effect at all. See: https://github.com/openwrt/openwrt/pull/12997#issuecomment-1612444072

Reply:
Replaced `reset-gpios` with
```
&gpio {
	phy-reset {
		gpio-hog;
		gpios = <11 GPIO_ACTIVE_HIGH>;
		output-high;
		line-name = "phy-reset";
	};
};
```

Reply:
`GPIO_ACTIVE_LOW` and `output-low` should be better. Low output means that phy should not be reset.

Reply:
Fixed
```
&gpio {
	phy-reset {
		gpio-hog;
		gpios = <11 GPIO_ACTIVE_LOW>;
		output-low;
		line-name = "phy-reset";
	};
};
```